src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs

```diff
@@ -0,0 +1,319 @@
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            HelpMessage = "The Database object to get backups for.")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlDatabaseModel InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Database Resource ID to get backups for.
+        /// </summary>
+        [Parameter(ParameterSetName = GetBackupByResourceIdSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            HelpMessage = "The Database Resource ID to get backups for.")]
+        [Parameter(ParameterSetName = GetBackupsByResourceIdSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            HelpMessage = "The Database Resource ID to get backups for.")]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Get the entities from the service
+        /// </summary>
+        /// <returns>The list of entities</returns>
+        protected override IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> GetEntity()
+        {
+            if (InputObject != null)
+            {
+                LocationName = InputObject.Location;
+                ServerName = InputObject.ServerName;
+                DatabaseName = InputObject.DatabaseName;
+            }
+            else if (!string.IsNullOrWhiteSpace(ResourceId))
+            {
+                string[] tokens = ResourceId.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+
+                ServerName = tokens[ServerNameIndex];
+                DatabaseName = tokens[DatabaseNameIndex];
+            }
+
+            return ModelAdapter.GetDatabaseLongTermRetentionBackups(
+                    LocationName,
+                    ServerName,
+                    DatabaseName,
+                    BackupName,
+                    OnlyLatestPerDatabase.IsPresent,
+                    DatabaseState);
+        }
+
+        /// <summary>
+        /// No user input to apply to model
+        /// </summary>
+        /// <param name="model">Model retrieved from service</param>
+        /// <returns>The model that was passed in</returns>
+        protected override IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> ApplyUserInputToModel(
+            IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> model)
+        {
+            return model;
+        }
+
+        /// <summary>
+        /// No changes to persist to server
+        /// </summary>
+        /// <param name="entity">The output of apply user input to model</param>
+        /// <returns>The input entity</returns>
+        protected override IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> PersistChanges(
+            IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> entity)
+        {
+            return entity;
+        }
+    }
```

Comment:
Should this cmdlet implement ExecuteCmdlet? Or does the base class cover the scenario entirely?

Reply:
No need here.